cmake/ConfigureCVC4.cmake

```diff
@@ -53,7 +54,7 @@ if(CVC4_WINDOWS_BUILD)
     add_c_cxx_flag(-pthread)
   endif()
 else()
-  check_symbol_exists(clock_gettime "time.h" HAVE_CLOCK_GETTIME)
+  check_library_exists(rt clock_gettime "time.h" HAVE_CLOCK_GETTIME)
```

Comment:
Hm, the documentation doesn't say that `check_library_exists` actually adds the library to the linking flags, so maybe we have to do something like:

```
check_symbol_exists(clock_gettime "time.h" HAVE_CLOCK_GETTIME)
if(NOT HAVE_CLOCK_GET_TIME)
  check_library_exists(rt clock_gettime "time.h" HAVE_CLOCK_GETTIME)
  find_library(RT_LIBRARIES NAMES rt)
endif()
```

And then `target_link_libraries(cvc4 ${RT_LIBRARIES})` in the appropriate places.

@mpreiner What do you think?

Reply:
Good point -- it does actually build with only check_library_exists but I'm not sure why.

Just out of curiosity, is this the documentation you checked? https://cmake.org/cmake/help/latest/module/CheckLibraryExists.html

Reply:
Yep, that's the one. Which OS was this on? With `make VERBOSE=1`, are you seeing the `-lrt` flag?

Reply:
CentOS. And huh, that’s very strange... I grepped for it in the verbose output and don’t see `-lrt`...but it builds and if I undo the change it complains about clock_gettime symbols..

Reply:
It could be that some other library includes rt. I'll take a look.